### PR TITLE
feat(w3c/headers): error if w3cid is missing for recTrack docs

### LINKED ISSUE #4055: Error or warning for a missing w3cid in respecConfig
Recent versions of pubrules checker consider it as an error if the `data-editor-id` attribute is missing for an editor. (I am not sure it complains for every specification type, though; it did it for me for a Rec.) 

This HTML attribute is generated from respec's [`w3cid` field in `respecConfig`](https://respec.org/docs/#person). However, that field is not marked as required in the respec documentation, nor does respec warn if it is missing. For editors making the connection between the pubrules error and a specific respec field may not be obvious…

A simple solution would be if respec reported either an error or at least a warning if that field is missing.

cc @msporny

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #4055

### CODE REVIEW

File: src/w3c/headers.js [hunk 1]
```diff
@@ -450,6 +450,16 @@ export async function run(conf) {
...
     const hint = docLink`Add one or more editors using the ${"[editors]"} configuration option.`;
     showError(msg, name, { hint });
+  } else if (conf.editors.length && conf.isRecTrack) {
+    // check that every editor has w3cid
+    conf.editors.forEach((editor, i) => {
+      if (editor.w3cid) return;
+      const msg = docLink`Editor ${
+        editor.name ? `"${editor.name}"` : `number ${i + 1}`
+      } is missing their ${"[w3cid]"}.`;
+      const hint = docLink`Please add a ${"[w3cid]"}. See ${"[`w3cid`]"} for instructions for how to retrieve it.`;
```
Comment (sidvishnoi) [inline src/w3c/headers.js:460]: Let's not link w3cid twice. Feels verbose.

File: tests/spec/w3c/headers-spec.js [hunk 2]
```diff
@@ -495,6 +496,23 @@ describe("W3C — Headers", () => {
...
+          group: "webapps",
+          github: "w3c/respec",
+          crEnd: "2019-01-01",
+        });
+        const doc = await makeRSDoc(ops, simpleSpecURL);
+        const errors = headerErrors(doc);
+        expect(errors).toHaveSize(1);
+        const [error] = errors;
+        expect(error.message).toContain("w3cid");
+        expect(error.message).toContain("bad person");
```
Comment (sidvishnoi) [inline tests/spec/w3c/headers-spec.js:513]: Umm?
```suggestion
```

File: src/w3c/headers.js (hunk 1 above)
Comment (marcoscaceres) [inline src/w3c/headers.js:460]:
```suggestion
      const hint = docLink`See ${"[`w3cid`]"} for instructions for how to retrieve it and add it.`;
```

File: tests/spec/w3c/headers-spec.js (hunk 2 above)
Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js:513]: I want "bad person", to make sure it the name is shown.

**Review by sidvishnoi — APPROVED** [review]

**Comment by marcoscaceres** [comment]
Yeah, I meant to update that too @xfq. However, I need to go over an update a bunch of stuff in headers.js to fix that. This solves the 80% case.

**Pull request merged**